lua/hbac/init.lua

```diff
@@ -82,17 +82,58 @@ M.setup = function(opts)
 end
 
 
-local function close_unused()
+M.close_unpinned = function()
 	local curbufnr = vim.api.nvim_get_current_buf()
 	local buflist = vim.api.nvim_list_bufs()
 	for _, bufnr in ipairs(buflist) do
-		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.persistant_buffers[bufnr] then
+		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.pinned_buffers[bufnr] then
 			vim.cmd('bd ' .. tostring(bufnr))
 		end
+		print("Hbac: Closed unpinned buffers")
 	end
 end
 
+
+M.pin = function(bufnr)
+  M.pin_buffer(bufnr)
+  print("Hbac: Buffer pinned")
+end
```

Comment:
I realize `pin` vs. `pin_buffer` may be confusing, please advise. The reason for this distinction is to only send the notification to the user when they explicitly call the `pin` command rather than during the automated pinning function.

Reply:
As mentioned in other thread pin_bufer should be a local func pin, that resolves naming conflict. Furthermore, if you agree to refactor it to local function toggle_pin, it can return a string message, which you can use to construct a notification or completely ignore when used internally in an autocommand